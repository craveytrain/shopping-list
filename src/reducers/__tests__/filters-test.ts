import reducer from 'reducers/filters';
import {
  visibilityFilter
} from 'actions/filters';
import * as deepFreeze from 'deep-freeze';

describe('filters reducer', () => {
  describe('visibilityFilter', () => {
    it('should set visibility to SHOW_ALL', () => {
      const initialState = {
        visibility: 'stuff'
      };

      deepFreeze(initialState);

      const nextState = reducer(initialState, visibilityFilter('SHOW_ALL'));

      expect(nextState).toMatchSnapshot();
    });

    it('should set visibility to SHOW_CHECKED', () => {
      const initialState = {
        visibility: 'stuff'
      };

      deepFreeze(initialState);

      const nextState = reducer(initialState, visibilityFilter('SHOW_CHECKED'));

      expect(nextState).toMatchSnapshot();
    });

    it('should set visibility to SHOW_UNCHECKED', () => {
      const initialState = {
        visibility: 'stuff'
      };

      deepFreeze(initialState);

      const nextState = reducer(initialState, visibilityFilter('SHOW_UNCHECKED'));

      expect(nextState).toMatchSnapshot();
    });
  });

  it('can be used with reduce', () => {
    const actionQueue = [
      visibilityFilter('SHOW_CHECKED')
    ];

    const finalState = actionQueue.reduce(reducer, []);

    expect(finalState).toMatchSnapshot();
  });
});
