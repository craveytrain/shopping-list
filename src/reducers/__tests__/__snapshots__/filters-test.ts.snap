// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters reducer can be used with reduce 1`] = `
Object {
  "visibility": "SHOW_CHECKED",
}
`;

exports[`filters reducer visibilityFilter should set visibility to SHOW_ALL 1`] = `
Object {
  "visibility": "SHOW_ALL",
}
`;

exports[`filters reducer visibilityFilter should set visibility to SHOW_CHECKED 1`] = `
Object {
  "visibility": "SHOW_CHECKED",
}
`;

exports[`filters reducer visibilityFilter should set visibility to SHOW_UNCHECKED 1`] = `
Object {
  "visibility": "SHOW_UNCHECKED",
}
`;
